src/Random/Distribution.h

```diff
@@ -0,0 +1,117 @@
+// Class Distribution
+//   This class can be used for creating a distribution object
+//   with custom pdf, cdf, and estimate function that is used for
+//   the sampling method.
+//
+
+#ifndef IPPL_DISTRIBUTION_H
+#define IPPL_DISTRIBUTION_H
+
+#include "Random/Utility.h"
+#include "Types/ViewTypes.h"
+
+namespace ippl {
+  namespace random {
+  
+    /*!
+     * @file Distribution.h
+     * @class Distribution
+    */
+  
+   /*!
+    * @ingroup Distribution
+    * @brief The class that represents a distribution.
+    * @tparam T Datatype.
+    * @tparam Dim Dimensionality of sample space.
+    * @tparam DimP Dimensionality of the parameter array.
+    * @tparam PDF Struct type for the PDF (Probability Density Function).
+    * @tparam CDF Struct type for the CDF (Cumulative Distribution Function).
+    * @tparam ESTIMATE Struct type for the ESTIMATE function.
+    */
+    template <typename T, unsigned Dim, unsigned DimP, typename PDF, typename CDF, typename ESTIMATE>
+    class Distribution {
+    public:
+        
+       /*!
+        * @param par_m An array of distribution parameters.
+        * @param pdf_m PDF of the distribution class as a member functor.
+        * @param cdf_m CDF of the distribution class as a member functor.
+        * @param estimate_m Estimate of the initial guess for the sampling method as a member functor.
+       */
+       T par_m[DimP];
+       PDF pdf_m;
+       CDF cdf_m;
+       ESTIMATE estimate_m;
```

Comment:
As these are public ... you do not need getter functions.

Reply:
Here, the get functions are written so that one does not need to pass parameters of distribution as an argument to compute pdf, cdf, estimate once they are initialized.